hy/compiler.py

```diff
@@ -1737,10 +1737,35 @@ def compile_require(self, expression):
         "unimport" it after we've completed `thing' so that we don't pollute
         other envs.
         """
-        expression.pop(0)
-        for entry in expression:
-            __import__(entry)  # Import it fo' them macros.
-            require(entry, self.module_name)
+        for entry in expression[1:]:
```

Comment:
They use the same syntax but do pretty different things because `import` actually emits code and `require` just changes the state of the compiler. I think trying to unify the implementations wouldn't make the compiler code much nicer.